python/rmm/rmm/pylibrmm/device_buffer.pyi

```diff
@@ -0,0 +1,62 @@
+# SPDX-FileCopyrightText: Copyright (c) 2019-2025, NVIDIA CORPORATION.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Optional, overload
+
+import numpy as np
+from numpy.typing import NDArray
+
+from rmm.pylibrmm.memory_resource import DeviceMemoryResource
+from rmm.pylibrmm.stream import Stream
+
+class DeviceBuffer:
+    mr: DeviceMemoryResource
+    stream: Stream
+
+    def __init__(
+        self,
+        *,
+        ptr: int = 0,
+        size: int = 0,
+        stream: Stream = ...,
+        mr: Optional[DeviceMemoryResource] = None,
+    ) -> None: ...
+    def __len__(self) -> int: ...
+    def __sizeof__(self) -> int: ...
+    def __bytes__(self) -> bytes: ...
+    @property
+    def nbytes(self) -> int: ...
+    @property
+    def ptr(self) -> int: ...
+    @property
+    def size(self) -> int: ...
+    @property
+    def __cuda_array_interface__(self) -> dict[str, Any]: ...
+    def prefetch(
+        self, device: Optional[int] = None, stream: Optional[Stream] = None
+    ) -> None: ...
+    def copy(self) -> DeviceBuffer: ...
+    def __copy__(self) -> DeviceBuffer: ...
+    @staticmethod
+    def to_device(b: bytes, stream: Stream = ...) -> DeviceBuffer: ...
+    @overload
+    def copy_to_host(self, stream: Stream = ...) -> NDArray[np.uint8]: ...
+    @overload
+    def copy_to_host(
+        self, ary: bytearray, stream: Stream = ...
+    ) -> bytearray: ...
+    def copy_from_host(self, ary: bytes, stream: Stream = ...) -> None: ...
```

Comment:
Check the type for `ary`. Probably buffer (https://docs.python.org/3/reference/datamodel.html#python-buffer-protocol)